### DISCUSSION TITLE
[ZK Stack] - Configuring multiple parallel Provers

### DISCUSSION THREAD

**Discussion by 6789tc** [zkSync-Community-Hub/zksync-developers#485] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [X] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.75.0

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

We are currently using ZK Stack to enable builders to spin up their ZK Rollups on Bitcoin via an intermediate EVM-compatible Layer 1 called BVM (https://bvm.network/).

We managed to get a testnet up and running (thanks to the excellent instructions) and have been conducting stress tests by generating numerous transactions. This has resulted in larger batches and slower proving times.

We believe that ZK Sync Era might have faced similar challenges and perhaps used multiple provers to parallelize the proving process.

Are there any instructions available for setting up multiple parallel provers?

### Expected Behavior

Instructions are available for setting up multiple parallel provers to speed up the proving process.

### Current Behavior

With a single prover, it would take a long time to generate zk proofs for a large batch.

### Repository Link (if applicable)

https://github.com/TrustlessComputer/zk-rollups

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by bxpana** [comment]
Thanks for the posting, we'll take some time to get back to this

**Reply by bxpana** [reply, marked as answer]
@6789tc parallelizing proofs for the same batch is not yet possible, but this is something we are actively working on for improving the system
